tasks/install_powershell.ps1

```diff
@@ -0,0 +1,50 @@
+[CmdletBinding()]
+Param(
+	[String]$version
+)
+# If an error is encountered, the script will stop instead of the default of "Continue"
+$ErrorActionPreference = "Stop"
+
+$arch = if ((Get-WmiObject Win32_OperatingSystem).OSArchitecture -match '^32') {"x86"} else {"x64"}
+
+if ($version) {
+    $msi_source = "http://downloads.puppetlabs.com/windows/puppet5/puppet-agent-${version}-${arch}.msi"
+}
+else {
+    $msi_source = "http://downloads.puppetlabs.com/windows/puppet5/puppet-agent-${arch}-latest.msi"
+}
+
+$msi_dest = Join-Path ([System.IO.Path]::GetTempPath()) "puppet-agent-$arch.msi"
+
+function DownloadPuppet {
+  Write-Verbose "Downloading the Puppet Agent for Puppet Enterprise on $env:COMPUTERNAME..."
+
+  $webclient = New-Object system.net.webclient
+
+  try {
+    $webclient.DownloadFile($msi_source,$msi_dest)
+  }
+  catch [System.Net.WebException] {
+    # If we can't find the msi, then we may not be configured correctly
+    if($_.Exception.Response.StatusCode -eq [system.net.httpstatuscode]::NotFound) {
+        Throw "Failed to download the Puppet Agent installer: $msi_source"
+    }
+    # Throw all other WebExceptions
+    Throw $_
+  }
+}
+
+function InstallPuppet {
+  $msiexec_args = "/qn /i $msi_dest"
```

Comment:
Additional Install arguments should parameterised.  e.g. Install logs for debug. 

* Also you're missing `RESTART=` otherwise you may get an unexpected reboot during a bolt run.

Reply:
Where do the MSI logs go if you don't specify anything?

Reply:
They don't log to a file.  You just the usual minimal logs in the eventlog.

Reply:
Current implementation will log the install to temp location. If the installation fails, the file (and MSI) are persisted. Upon successful installation both the log and the MSI are deleted.